### PR TITLE
k_work conversion for network related components

### PR DESCRIPTION
**Pull request by jukkar**
This is related to #33104 and takes relevant patches from #33924. I did slight changes to some of the commits from #33824, those ones have my signed-off-by added.
Only minimal testing done for those components that I could, but overall the original rote conversion by @pabigot looked good.

### CODE REVIEW

File: drivers/ieee802154/ieee802154_dw1000.c
Comment (rlubos) [inline drivers/ieee802154/ieee802154_dw1000.c:1610]: Not really related to the rework, but his work queue seems a bit suspicious, it doesn't actually seem to be used anywhere in the driver. CC @jfischer-no

**Review by pabigot — COMMENTED** [review]
I can't approve this officially since the commits are mostly from me, but I see no problem with these changes.

File: drivers/ethernet/eth_sam_gmac.c
Comment (rlubos) [inline drivers/ethernet/eth_sam_gmac.c:1975]: nit: there are quite a few places in this PR with broken alignment after the API change, like here, probably worth pushing the changes through some formatter.

**Review by rlubos — COMMENTED** [review]
The API change looks reasonable to me, there are some style issues worth fixing I guess.

File: drivers/ethernet/eth_sam_gmac.c
Comment (jukkar) [reply drivers/ethernet/eth_sam_gmac.c:1975]: > there are quite a few places in this PR with broken alignment after the API change, like here, probably worth pushing the changes through some formatter.

Yeah, there are now these style issues in the code. I went through the changes in the `subsys/net/*` and manually changed those. I did not check all the changes in this PR (especially in the drivers/) as there was just so much of it. Using formatter could work except that it can change lines that we not changing in this k_work conversion, so I did not use a formatter. I think this work can be done in a separate PR if really needed.

**Comment by jukkar** [comment]
> I can't approve this officially since the commits are mostly from me, but I see no problem with these changes.

Sure you can as you are not the sender of this PR. This one needs approvers and nobody seems to willing to accept the changes.

File: drivers/ethernet/eth_sam_gmac.c
Comment (rlubos) [reply drivers/ethernet/eth_sam_gmac.c:1975]: > Using formatter could work except that it can change lines that we not changing in this k_work conversion, so I did not use a formatter.

I typically use `git-clang-format HEAD~1` to apply the formatting only to the changes from the most recent commit, this combines well with an interactive rebase. But I now noticed that while it fixes the alignment issues I spotted it also introduce some others (like for instance changing the commit alignment from tabs to spaces which in effect doesn't look good with other lines aligned with tabs). As it's really not that important I guess it could indeed be fixed at some point in the future in a more comprehensive approach.

**Review by rlubos — APPROVED** [review]

File: drivers/ieee802154/ieee802154_dw1000.c
Comment (jfischer-no) [reply drivers/ieee802154/ieee802154_dw1000.c:1610]: I think k_work_submit should actually be `k_work_submit_to_queue(&dwt_work_queue, &ctx->irq_cb_work);` I will look through my WIP branches.

File: drivers/ethernet/eth_sam_gmac.c
Comment (jukkar) [reply drivers/ethernet/eth_sam_gmac.c:1975]: > use `git-clang-format HEAD~1` to apply the formatting only to the changes from the most recent commit

Thanks for the hint, I was not aware of this tool.

Comment (jukkar) [reply drivers/ethernet/eth_sam_gmac.c:1975]: > As it's really not that important I guess it could indeed be fixed at some point in the future in a more comprehensive approach.

I will revisit this PR and update the code. I did not realize there was a tool for doing the work and it looked a lot of work to go through the patches manually.

**Comment by jukkar** [comment]
Fixed the code indentation

**Comment by pabigot** [comment]
@rlubos @jukkar If people are interested in using `clang-format` somebody could take over #27058.  I think the list of FOR-EACH macros probably needs to be updated.

**Review by jfischer-no — APPROVED** [review]

**Pull request merged**
